specs/shared_inventory_openapi.yaml

```diff
@@ -303,151 +944,248 @@ components:
             (scope and select) resources
           type: array
           items:
-            $ref: "#/components/schemas/ResourceTag"
-      discriminator:
-        propertyName: ResourceType
+            $ref: '#/components/schemas/ResourceTag'
 
-    k8sCluster:
+    Resource:
       allOf:
         - $ref: "#/components/schemas/CommonAttributes"
         - type: object
-          description: Definition of k8s cluster resource  type
+          description: Attributes shared by all items in inventory
           required:
-            - ResourceType
+            - DisplayName
+            - Data
+            - ReporterData
+          properties:
+            Data:
+              oneOf:
+                - $ref: '#/components/schemas/K8sClusterDetail'
+                - $ref: '#/components/schemas/RootPolicyDetail'
+                - $ref: '#/components/schemas/IntermediatePolicyDetail'
+    
+    K8sCluster:
```

Comment:
Lets populate a real example - if it makese sense. Would be happy to provide real data for this if it helps